src/dqe_idx.erl

```diff
@@ -227,6 +228,28 @@ metrics(Collection) ->
 
 %%--------------------------------------------------------------------
 %% @doc
+%% Returns a list of metric path suffixes of `Depth' that are prefixed
+%% by the given probe `Prefix', which can also be empty.
+%%
+%% For example:
+%% metrics(<<"collection">>, <<>>, 1) -> [<<"base">>].
+%% metrics(<<"collection">>, <<"base">>, 1) -> [<<"cpu">>].
+%% metrics(<<"collection">>, <<>>, 2) -> [[<<"base">>.<<"cpu">>]].
+%%
+%% Actual results are binary-encoded in `dproto' encoding.
+%% @end
+%%--------------------------------------------------------------------
+-spec metrics(Collection::collection(),
+              Prefix::metric(),
+              Depth::pos_integer()) ->
+                    {ok, [Metric::metric()]} |
```

Comment:
One other very simple thing just came to mind. Lets ignore the _Ddb driver as it stubs out most of the functions anyway.

The postgres driver now returns lists for queries, it's it's native formate, so we know that at least once we'll have to do the list to binary conversion. When making list the native format for dqe_idx we can delay this translation towards the end of the logic and in some cases skip it alltogether. While using the binary representation we would at least do the conversion once and in some cases do it two or three times (list -> binary -> list -> binary).